bindings/ocaml/src/dune

```diff
@@ -16,20 +16,20 @@
 ; under the License.
 
 (rule
- (targets libopendal_core.a dllopendal_core.so)
+ (targets libopendalCore.a dllopendalCore.so)
  (deps
   (glob_files *.rs))
  (action
   (progn
    (run sh -c "cd %{project_root}/../.. && cargo build --release")
    (run sh -c
-     "mv %{project_root}/../../../../target/release/libopendal_ocaml.so ./dllopendal_core.so 2> /dev/null || mv %{project_root}/../../../../target/release/libopendal_ocaml.dylib ./dllopendal_core.so")
+     "mv %{project_root}/../../../../target/release/libopendal_ocaml.so ./dllopendalCore.so 2> /dev/null || mv %{project_root}/../../../../target/release/libopendal_ocaml.dylib ./dllopendalCore.so")
    (run mv %{project_root}/../../../../target/release/libopendal_ocaml.a
-     libopendal_core.a))))
+     libopendalCore.a))))
 
 (library
- (name opendal_core)
- (public_name opendal.opendal_core)
- (foreign_archives opendal_core)
+ (name opendalCore)
```

Comment:
Sorry for bringing up naming issues again. I want to clarify that I don't have a preference for a specific naming style. What I want is consistency in the naming conventions used.

For example:

- `opendal_core`
- `OpendalCore`

Instead of

- `libopendalCore.a`
- `dllopendalCore.so`
- `opendalCore`

Given we have some limitations in the project, maybe the closest one is

- `Opendal_Core` for places name must start with upper case
- `opendal_core` for places name must start with lower case

What do you think?

Reply:
If we are in dune or the file name starts with a lowercase letter such as `opendal_core`, it will be automatically capitalized as `Opendal_core` in the ocaml module.

One naming scheme is to use all lowercase files, connect with `_`,On the module, it starts with uppercase, and other lowercase styles,
Janestreet adopted this style, in Base, Core, we can see many examples of this. like [ref](https://github.com/janestreet/base/blob/master/src/comparable_intf.ml#L8)

Similarly, there are examples of using all capital letters in the standard library, such as [ref](https://github.com/ocaml/ocaml/blob/trunk/stdlib/moreLabels.mli#L697) This module is connected with capital letters.

The `Opendal_Core` style doesn't seem to be picked up from the standard library.

As we all know, naming is a complicated problem, and there seems to be no definite standard in ocaml.

In our project, the special thing is that the prefixes `lib` and `dll` of `.so` and `.a` cannot be modified. We also cannot use the naming method of `lib_opendal_core.so` because the name cannot start with `_` start.

So we have the following naming methods to choose from

1.   dune&File: `opendal_core`, module: `Opendal_core`, dll: `libopendal_core.so`

2.   dune&File: `opendalCore`, module: `OpendalCore`, dll: `libopendalCore.so`

3.  dune&File: `OpendalCore`, module: `OpendalCore`, dll: `libOpendalCore.so`

For me personally, 1 or 2 can find more examples from the standard library, it seems suitable for selection

Reply:
> For me personally, 1 or 2 can find more examples from the standard library, it seems suitable for selection

Thanks for the research! Let's use option `1`.